osbenchmark/aggregator.py

```diff
@@ -0,0 +1,216 @@
+from typing import Any, Dict, List, Union
+import uuid
+
+from osbenchmark.metrics import FileTestExecutionStore
+from osbenchmark import metrics, workload, config
+
+class Aggregator:
+    def __init__(self, cfg, test_executions_dict):
+        self.config = cfg
+        self.test_executions = test_executions_dict
+        self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
+        self.accumulated_iterations: Dict[str, int] = {}
+
+    # count iterations for each operation in the workload
+    def iterations(self) -> None:
+        loaded_workload = workload.load_workload(self.config)
+        for task in loaded_workload.test_procedures:
+            for operation in task.schedule:
+                operation_name = operation.name
+                iterations = operation.iterations or 1
+                self.accumulated_iterations.setdefault(operation_name, 0)
+                self.accumulated_iterations[operation_name] += iterations
+
+    # accumulate metrics for each task from test execution results
+    def results(self, test_execution: Any) -> None:
+        for item in test_execution.results.get("op_metrics", []):
+            task = item.get("task", "")
+            self.accumulated_results.setdefault(task, {})
+            for metric in ["throughput", "latency", "service_time", "client_processing_time", "processing_time", "error_rate", "duration"]:
```

Comment:
Let's store this list as a class variable

Reply:
Done!